pkg/controller/vitesskeyspace/reconcile_resharding.go

```diff
@@ -136,3 +156,33 @@ func (r *reconcileHandler) reconcileResharding(ctx context.Context) (reconcile.R
 
 	return resultBuilder.Result()
 }
+
+func (r reconcileHandler) shardsRowCount(ctx context.Context, shardNames []string) (uint64, error) {
+	var rowCount uint64
+	for _, shardName := range shardNames {
+		tabletMap, err := r.ts.GetTabletMapForShard(ctx, r.vtk.Spec.Name, shardName)
+		if err != nil {
+			return 0, err
+		}
+		// Find the master and get data size from it.
+		var masterTabletAlias *topodata.TabletAlias
+		for _, tabletInfo := range tabletMap {
+			if tabletInfo.Type == topodata.TabletType_MASTER {
+				masterTabletAlias = tabletInfo.Alias
+				break
+			}
+		}
+		if masterTabletAlias == nil {
+			return 0, fmt.Errorf("could not find master tablet alias for determining row count")
+		}
+		schema, err := r.wr.GetSchema(ctx, masterTabletAlias, make([]string, 0), make([]string, 0), false)
```

Comment:
Do we really need to make string arrays, or can we just pass `nil, nil`? If we do need to allocate something, the more common way to make an empty string slice is `[]string{}`.

Reply:
wasn't sure if you can call `len()` on nil, but just tried in go playground and you can. Refactoring to `nil, nil`.

Reply:
Done.